plugins/communications-app/BodyForm/index.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { Form } from '@openedx/paragon';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import TextEditor from '@communications-app/src/components/bulk-email-tool/text-editor/TextEditor';
+import { useSelector, useDispatch } from '@communications-app/src/components/bulk-email-tool/bulk-email-form/BuildEmailFormExtensible/context';
+import { actionCreators as formActions } from '@communications-app/src/components/bulk-email-tool/bulk-email-form/BuildEmailFormExtensible/context/reducer';
+
+import messages from './messages';
+
+const BodyForm = () => {
+  const intl = useIntl();
+  const formData = useSelector((state) => state.form);
+  const dispatch = useDispatch();
+  const { body, isFormSubmitted = false } = formData;
+
+  const handleChangeTextEditor = (value) => {
+    dispatch(formActions.updateForm({ body: value }));
+  };
+
+  const isBodyValid = body.length > 0;
```

Comment:
In the previous implementation, the same validation code is used to show/hide error messages *and* prevent `POST`ing.

https://github.com/openedx/frontend-app-communications/blob/6dd835d63f106c6e1bf7a02de16d3047a5bee558/src/components/bulk-email-tool/bulk-email-form/BulkEmailForm.jsx#L153-L165

https://github.com/openedx/frontend-app-communications/blob/6dd835d63f106c6e1bf7a02de16d3047a5bee558/src/components/bulk-email-tool/bulk-email-form/BulkEmailForm.jsx#L297

In this PR the logic has been split up and will need to be updated in multiple places if it ever changes. Have you considered ways to be able to utilize the same validation code for displaying errors and for preventing `POST`?

Reply:
I was wondering about use  [yup](https://github.com/jquense/yup) which allow to have an scheme to validate the fields that we need and then show the errors in the please required but I do not implemented here since the validations are only for the length of the fileds

Reply:
I'd prefer to avoid adding another library dependency, I was thinking something more along the lines of having the existing validation logic live in a file that we can import from in multiple places.

Reply:
Having account that we are using multiple plugins where place do you recommend? I wouldn't like to do it in the MFE now, in another plugin?